utils/utils_test.go

```diff
@@ -1,10 +1,195 @@
+	}
+
+	for _, st := range subtests {
+		t.Run(st.name, func(t *testing.T) {
+			batches := BatchSliceOfStrings(context.Background(), st.keys, st.count)
+			require.NotNil(t, batches)
+
+			for _, expected := range st.output {
+				select {
+				case actual, ok := <-batches:
+					require.True(t, ok, "receiving should return a value")
+					require.Equal(t, expected, actual)
+				case <-time.After(10 * time.Millisecond):
+					require.Fail(t, "receiving should not block")
+				}
+			}
+
+			select {
+			case _, ok := <-batches:
+				require.False(t, ok, "receiving from channel should not return anything")
+			case <-time.After(10 * time.Millisecond):
+				require.Fail(t, "receiving should not block")
+			}
+		})
+	}
+}
+
+func TestChecksum(t *testing.T) {
+	subtests := []struct {
+		name  string
+		input any
+	}{
+		{"empty_string", ""},
+		{"empty_bytes", []byte(nil)},
+		{"space_string", " "},
+		{"space_bytes", []byte(" ")},
+	}
+
+	for _, st := range subtests {
+		t.Run(st.name, func(t *testing.T) {
+			actual := Checksum(st.input)
+
+			require.Len(t, actual, 20)
+			require.NotEqual(t, make([]byte, 20), actual)
+		})
+	}
+}
+
+func TestIsDeadlock(t *testing.T) {
+	msg := "Unsuccessful attempt of confusing the tested code."
+	code := [5]byte{0, 23, 42, 77, 255}
+
+	subtests := []struct {
+		name   string
+		input  error
+		output bool
+	}{
+		{"nil", nil, false},
+		{"deadline", context.DeadlineExceeded, false},
+		{"mysql1204", &mysql.MySQLError{Number: 1204}, false},
+		{"mysql1205", &mysql.MySQLError{Number: 1205}, true},
+		{"mysql1205_with_crap", &mysql.MySQLError{Number: 1205, SQLState: code, Message: msg}, true},
+		{"mysql1206", &mysql.MySQLError{Number: 1206}, false},
+		{"mysql1212", &mysql.MySQLError{Number: 1212}, false},
+		{"mysql1213", &mysql.MySQLError{Number: 1213}, true},
+		{"mysql1213_with_crap", &mysql.MySQLError{Number: 1213, SQLState: code, Message: msg}, true},
+		{"mysql1214", &mysql.MySQLError{Number: 1214}, false},
+		{"postgres40000", &pq.Error{Code: "40000"}, false},
+		{"postgres40001", &pq.Error{Code: "40001"}, true},
+		{"postgres40001_with_crap", &pq.Error{Code: "40001", Message: msg}, true},
+		{"postgres40002", &pq.Error{Code: "40002"}, false},
+		{"postgres40P01", &pq.Error{Code: "40P01"}, true},
+		{"postgres40P01_with_crap", &pq.Error{Code: "40P01", Message: msg}, true},
+	}
+
+	for _, st := range subtests {
+		t.Run(st.name, func(t *testing.T) {
+			require.Equal(t, st.output, IsDeadlock(st.input))
+		})
+	}
+}
+
+func TestEllipsize(t *testing.T) {
+	subtests := []struct {
+		name   string
+		s      string
+		limit  int
+		output string
+	}{
+		{"negative", "", -1, "..."},
+		{"empty", "", 0, ""},
+		{"shorter", " ", 2, " "},
+		{"equal", " ", 1, " "},
+		{"longer", " ", 0, "..."},
+		{"unicode", "äöüß€", 4, "ä..."},
+	}
+
+	for _, st := range subtests {
+		t.Run(st.name, func(t *testing.T) {
+			require.Equal(t, st.output, Ellipsize(st.s, st.limit))
+		})
+	}
+}
+
+func TestMaxInt(t *testing.T) {
```

Comment:
I would advise against testing this helper function. Furthermore, I would argue for removing it, as it has landed in the Go standard library a while ago with 1.21.0 as [`builtin.max`](https://pkg.go.dev/builtin#max).

Reply:
#65